src/include/mpir_gpu.h

```diff
@@ -105,6 +105,15 @@ MPL_STATIC_INLINE_PREFIX bool MPIR_GPU_query_pointer_is_dev(const void *ptr)
     return false;
 }
 
+MPL_STATIC_INLINE_PREFIX bool MPIR_GPU_query_pointer_is_strict_dev(const void *ptr)
+{
+    if (ENABLE_GPU && ptr != NULL) {
+        return MPL_gpu_query_pointer_is_strict_dev(ptr, NULL);
+    }
+
+    return false;
+}
+
```

Comment:
Should this be removed?

Reply:
I think we need to keep this because it is called during `win_allgather` where there are no guards for whether or not GPUs are enabled. `MPL_gpu_query_pointer_is_strict_dev` would have a hard failure for ZE if it was called without calling `MPL_gpu_init` first

Reply:
Oh, my mistake. I thought this was in the MPL header.

Reply:
Okay, the next question is why not all paths call this function?

Reply:
That I will fix :)

Reply:
Done.

Seems we might need to do a similar cleanup for calls to `MPL_gpu_query_pointer_is_dev`; that can be done in a separate PR